x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/execution/PlanExecutor.java

```diff
@@ -77,15 +77,17 @@ public void esql(
         );
         QueryMetric clientId = QueryMetric.fromString("rest");
         metrics.total(clientId);
-        session.execute(request, executionInfo, planRunner, wrap(x -> {
+
+        ActionListener<Result> executeListener = wrap(x -> {
             planningMetricsManager.publish(planningMetrics, true);
             listener.onResponse(x);
         }, ex -> {
             // TODO when we decide if we will differentiate Kibana from REST, this String value will likely come from the request
             metrics.failed(clientId);
             planningMetricsManager.publish(planningMetrics, false);
             listener.onFailure(ex);
-        }));
+        });
+        ActionListener.run(executeListener, l -> session.execute(request, executionInfo, planRunner, l));
```

Comment:
I'm not super happy with this double wrapping but if we don't do it, we can't capture a lot of planning-stage exceptions.

Reply:
At a minimum can we document why it's needed?